api/src/services/script-runner.service.ts

```diff
@@ -28,34 +31,543 @@ export class ScriptRunnerService {
+          );
+        }
+        let applicationDropOffAddress;
+        if (listing['application_drop_off_address_id']) {
+          applicationDropOffAddress = await client.$queryRawUnsafe(
+            `SELECT * FROM address WHERE id = '${listing['application_drop_off_address_id']}'`,
+          );
+        }
+        let applicationMailingAddress;
+        if (listing['application_mailing_address_id']) {
+          applicationMailingAddress = await client.$queryRawUnsafe(
+            `SELECT * FROM address WHERE id = '${listing['application_mailing_address_id']}'`,
+          );
+        }
+        let reservedCommunityType;
+        if (listing['reserved_community_type_id']) {
+          reservedCommunityType = await client.$queryRawUnsafe(
+            `SELECT * FROM reserved_community_types WHERE id = '${listing['reserved_community_type_id']}'`,
+          );
+          if (reservedCommunityType.length) {
+            const reservedCommunityTypes =
+              await this.prisma.reservedCommunityTypes.findMany();
+            const name = reservedCommunityType[0]['name'];
+            const foundReservedCommunityType = reservedCommunityTypes.find(
+              (communityType) =>
+                communityType.name === name ||
+                (name === 'senior' && communityType.name === 'senior55'),
+            );
+            if (foundReservedCommunityType) {
+              reservedCommunityType = foundReservedCommunityType;
+            }
+          }
+        }
+
+        const createdListing = await this.prisma.listings.create({
+          data: {
+            id: listing['id'],
+            listingsBuildingAddress: buildingAddress?.length
+              ? {
+                  create: this.createAddress(
+                    buildingAddress,
+                    dataTransferDTO.jurisdiction,
+                  ),
+                }
+              : undefined,
+            listingsLeasingAgentAddress: leasingAgentAddress?.length
+              ? {
+                  create: this.createAddress(leasingAgentAddress),
+                }
+              : undefined,
+            listingsApplicationPickUpAddress: applicationPickUpAddress?.length
+              ? { create: this.createAddress(applicationPickUpAddress) }
+              : undefined,
+            listingsApplicationDropOffAddress: applicationDropOffAddress?.length
+              ? { create: this.createAddress(applicationDropOffAddress) }
+              : undefined,
+            listingsApplicationMailingAddress: applicationMailingAddress?.length
+              ? { create: this.createAddress(applicationMailingAddress) }
+              : undefined,
+            jurisdictions: {
+              connect: {
+                id: doorwayJurisdiction.id,
+              },
+            },
+            reservedCommunityTypes: reservedCommunityType
+              ? {
+                  connectOrCreate: {
```

Comment:
since autosync reserved community types I think this can be a `connect` instead

Reply:
Good catch. I forgot to update this after getting confirmation from Product that we don't need to transfer any community types